src/Board.cpp

```diff
@@ -1,30 +1,60 @@
 #include "Board.hpp"
 #include "json.hpp"
-//#include "JSONFileReader.hpp"
 #include <fstream>
+#include <algorithm>
+#include "DisplayBoard.hpp"
 
-Board::Board(int rowNumber,
-             int colNumber,
+Board::Board(const unsigned int rowNumber,
+             const unsigned int colNumber,
              const Lines& rows,
              const Lines& cols) noexcept
     : rowNumber_(rowNumber),
       colNumber_(colNumber),
       rows_(rows),
       cols_(cols)
 {
-    Line singleRow;
-    singleRow.assign(colNumber_, 0);
+    BLine singleRow;
+    singleRow.assign(colNumber_, FieldsEnum::UNKNOWN);
     board_.assign(rowNumber_, singleRow);
 }
 
 Board::~Board() {}
 
-int Board::getRowsNumber() const
+unsigned int Board::getRowsNumber() const
 {
     return rowNumber_;
 }
 
-int Board::getColsNumber() const
+unsigned int Board::getColsNumber() const
 {
     return colNumber_;
 }
+
+BLines Board::getBoardLines() const
+{
+    return board_;
+}
+
+unsigned int Board::findLongestVectorInLines(const Lines& v) const
```

Comment:
`getLogestLineLength` sounds like a better name. It describes exactly what this function does.
I would expect that find function return an iterator.

The same for below function names.

Reply:
done